### ISSUE TITLE
bug: `encodeAbiParameters` improperly encodes emojis

### ISSUE THREAD

**Issue by ardislu** [wagmi-dev/viem#134]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.1.1

### Current Behavior

It looks like `encodeAbiParameters` doesn't encode certain emojis properly. I picked these: `👨‍👨‍👦‍👦🏴‍☠️` to test because these emojis consist of multiple Unicode code points but I'm not sure what exactly triggers the issue.

### Expected Behavior

All Unicode characters should be encoded properly.

### Steps To Reproduce

Encode the string:

```javascript
import { AbiCoder } from 'ethers';
import { encodeAbiParameters, parseAbiParameters } from 'viem';
const ethersCoder = new AbiCoder();

const types = ['string'];
const objects = ['👨‍👨‍👦‍👦🏴‍☠️'];

const ethersResult = ethersCoder.encode(types, objects);
console.log(ethersResult);
// 0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000026f09f91a8e2808df09f91a8e2808df09f91a6e2808df09f91a6f09f8fb4e2808de298a0efb88f0000000000000000000000000000000000000000000000000000

const viemResult = encodeAbiParameters(parseAbiParameters(types), objects);
console.log(viemResult);
// 0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000010d83ddc68200dd83ddc68200dd83ddc66200dd83ddc66d83cdff4200d2620fe0f
```

Then check it in the web browser DevTools console:

```javascript
function decodeUTF8(hex) {
  return new TextDecoder().decode(Uint8Array.from(hex.substring(2).match(/.{2}/g), v => parseInt(v, 16)));
}

// From ethersResult
decodeUTF8('26f09f91a8e2808df09f91a8e2808df09f91a6e2808df09f91a6f09f8fb4e2808de298a0efb88f');
// '👨‍👨‍👦‍👦🏴‍☠️'

// From viemResult
decodeUTF8('10d83ddc68200dd83ddc68200dd83ddc66200dd83ddc66d83cdff4200d2620fe0f');
// '�=�h \r�=�h \r�=�f \r�=�f�<�� \r& �\x0F'
```

### Link to Minimal Reproducible Example (StackBlitz, CodeSandbox, GitHub repo etc.)

https://github.com/ardislu/eth-abi-coder-comparison

**Comment by jxom** [comment]
Fix incoming! Thanks.

**Comment by jxom** [comment]
Released at `0.1.3`.
